KIALI-1617 Add VS/DR validation list

xeviknal opened the pull request:
List all the validations performed by kiali on Destination Rules and Virtual Services.

![screenshot of kiali 1](https://user-images.githubusercontent.com/613814/46811629-6e48d480-cd73-11e8-8ed8-7b4e8074395b.png)

## Column description:
Validation message: Message that is shown in the ACEditor
Severity: error/warning
Description: broader description about the validation
Source: link to source code
Example: link to a yaml that triggers the validation explained

abonas commented on content/documentation/overview/index.adoc:
```diff
@@ -115,6 +116,65 @@ Kiali will highlight configuration errors.
 
+==== Validations performed
+[cols="2,1,2,1,1", options="header"]
+.List of destination rule validations
```
is this the list of Kiali specific validations? the ones that are *not* done by Galley? or list of all validations together?

xeviknal replied:
they are the one not done by Galley, the specific ones created in Kiali.

abonas replied:
I would specify this in the documentation so it is clear these are "extra" ones that do not exist in Istio

abonas replied:
other than that, it is LGTM

xeviknal commented:
@lucasponce, I added the deprecated validations (same as kiali/kiali#1668)
@abonas I also added a little comment at the beginning of the section. Do you thing is clear enough?

abonas commented on content/documentation/overview/index.adoc:
```diff
@@ -115,6 +117,92 @@ Kiali will highlight configuration errors.
 
+==== Validations performed
+
+In this section, there are listed all the validations that Kiali perform for all the Istio Configs. Most of these validations are extra to the existent ones performed by Galley (excepting those marked as deprecated).
```
How about:
"This section lists all the validations that Kiali performs on all Istio configurations. Most of these validations are done in addition to/on top of the existing ones performed by Istio's Galley component (except those marked as deprecated)"

xeviknal commented:
@abonas comments approached :) thanks for the fine-tuning!
